app/lib/hooks/useA11yErrorAnnouncement.tsx

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from 'react';
+import { AccessibilityInfo } from 'react-native';
+
+interface IUseA11yErrorAnnouncement {
+	errorMessage: string | undefined;
+}
+
+const useA11yErrorAnnouncement = ({ errorMessage }: IUseA11yErrorAnnouncement) => {
+	const [errorAnnounced, setErrorAnnounced] = useState(false);
+
+	useEffect(() => {
```

Comment:
Since it looks like we're going to be using this on all forms, I think we should check how performant this is.

- It doesn't have a return, so I'm not sure it triggers a re-render on ReportUserView when `errorAnnounced` changes. I couldn't get what makes `errorAnnounced` change.
- I think we could remove `useEffect` and use `usePrevious` instead, which makes use of a ref
  - We would be comparing `errorMessage` and `previousErrorMessage`
  - `errorAnnounced` could also be a ref
  - `const shouldAnnounce = errorMessage && errorMessage !== previousMessageRef.current && !errorAnnounced;` <- This would be the first if condition
- What if the user changes the input's content too fast? Is it going to announce several times? Perhaps we should do a debounce?